src/libponyrt/actor/actor.c

```diff
@@ -281,15 +281,17 @@ static void try_gc(pony_ctx_t* ctx, pony_actor_t* actor)
 #endif
 }
 
-static void unblock(pony_actor_t* actor)
+static void maybe_unblock(pony_actor_t* actor)
 {
-  if(has_internal_flag(actor, ACTOR_FLAG_BLOCKED_SENT)) {
-    unset_internal_flag(actor, ACTOR_FLAG_BLOCKED | ACTOR_FLAG_BLOCKED_SENT);
-    ponyint_cycle_unblock(actor);
-  } else {
-    unset_internal_flag(actor, ACTOR_FLAG_BLOCKED);
+  if(has_internal_flag(actor, ACTOR_FLAG_BLOCKED)) {
+    if(has_internal_flag(actor, ACTOR_FLAG_BLOCKED_SENT)) {
```

Comment:
We discussed a bit on the sync call. I had a doubt that I expressed to Sean, and he suggested I write a comment on the ticket.

My thought is that if we've sent a blocked message to the cycle detector, we probably want to ensure that we don't neglect to send an unblock message to it. I'm concerned that this change, the way it's written, may have the effect of sometimes neglecting to send this message in cases where it's needed.

However, you and Sean know more about this than me, so I could easily be wrong here.

Reply:
> may have the effect of sometimes neglecting to send this message in cases where it's needed.

can you please elaborate what scenario you're referring to as i'm not following?